src/main/java/com/saucelabs/bamboo/sod/action/BuildConfigurator.java

```diff
@@ -120,8 +120,7 @@ public void println(String x) {
         SauceTunnelManager sauceTunnelManager = SauceConnectFourManagerSingleton.getSauceConnectFourTunnelManager();
         String options = getResolvedOptions(config.getSauceConnectOptions());
         if (config.useGeneratedTunnelIdentifier()) {
-            String tunnelIdentifier = generateTunnelIdentifier(buildContext.getPlanName());
-            customVariableContext.addCustomData(SODKeys.TEMP_TUNNEL_ID, tunnelIdentifier);
+            String tunnelIdentifier = customVariableContext.getVariables(buildContext).get(SODKeys.TUNNEL_ID);
             options = "--tunnel-identifier " + tunnelIdentifier + " " + options;
```

Comment:
What if you use getResolvedOptions? Move it before options gets evaluated?

On Wed, May 18, 2016 at 2:24 PM, Moiz Virani <notifications@github.com>
wrote:

> In src/main/java/com/saucelabs/bamboo/sod/action/BuildConfigurator.java
> <https://github.com/saucelabs/bamboo_sauce/pull/41#discussion_r63785090>:
>
> > @@ -120,8 +120,7 @@ public void println(String x) {
> >          SauceTunnelManager sauceTunnelManager = SauceConnectFourManagerSingleton.getSauceConnectFourTunnelManager();
> >          String options = getResolvedOptions(config.getSauceConnectOptions());
> >          if (config.useGeneratedTunnelIdentifier()) {
> > -            String tunnelIdentifier = generateTunnelIdentifier(buildContext.getPlanName());
> > -            customVariableContext.addCustomData(SODKeys.TEMP_TUNNEL_ID, tunnelIdentifier);
> > +            String tunnelIdentifier = customVariableContext.getVariables(buildContext).get(SODKeys.TUNNEL_ID);
> >              options = "--tunnel-identifier " + tunnelIdentifier + " " + options;
>
> nope unfortunately this does not work Command line arguments:
> /Users/moiz/sc-4.3.14-osx/bin/sc -u halkeye -k **** -P 4445
> --tunnel-identifier ${bamboo.TUNNEL_IDENTIFIER}
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/saucelabs/bamboo_sauce/pull/41/files/1a43211c2a6d835b8ba1993f2eef01f11955eba6#r63785090>
>



-- 


*Gavin MoganSoftware Developer*

Sauce Labs <http://saucelabs.com/> | Sauce blog <http://sauceio.com/>

Talk to us anytime:  <https://twitter.com/saucelabs>
<https://www.linkedin.com/company/sauce-labs>
<https://plus.google.com/+SauceLabs_Official/posts>
<https://www.facebook.com/saucelabs>